### PR TITLE
strategy_standard_sale_for_fixed_price.cairo converted to cairo1

### LINKED ISSUE #7: Implement Functions' Body for  StrategyStandardSaleForFixedPrice contract
Implement the functions' body for the [StrategyStandardSaleForFixedPrice](https://github.com/Flex-NFT-Marketplace/Flex-Marketplace-Contract/blob/main/flex_marketplace/src/marketplace/strategy_standard_sale_for_fixed_price.cairo) contract. For reference and consistency, the implementation should align with the existing Cairo0 [`StrategyStandardSaleForFixedPrice.cairo`](https://github.com/Flex-NFT-Marketplace/Flex-Marketplace-Contract/blob/main/contracts/marketplace/StrategyStandardSaleForFixedPrice.cairo) contract.

Dependencies:

scarb v2.4.0
snforge 0.13.0

Requirements:

Ensure that the code passes both `scarb fmt` and `scarb build` commands without errors, ensuring proper formatting and successful build.

### PR DESCRIPTION
**Pull request by AniDhumal**
Resolves #7

### CODE REVIEW

**Comment by AniDhumal** [comment]
@ametel01 @and1vo Can you review this please?

File: flex_marketplace/src/marketplace/strategy_standard_sale_for_fixed_price.cairo [hunk 1]
```diff
@@ -1,30 +1,32 @@
...
     ) -> (bool, u256, u128);
     fn can_execute_taker_bid(
         self: @TState, taker_bid: TakerOrder, maker_ask: MakerOrder
     ) -> (bool, u256, u128);
+    fn upgrade(ref self: TState, impl_hash: ClassHash);
 }
 
 #[starknet::contract]
 mod StrategyStandardSaleForFixedPrice {
+    use core::option::OptionTrait;
```
Comment (ametel01) [inline flex_marketplace/src/marketplace/strategy_standard_sale_for_fixed_price.cairo:21]: no need to import, it's already in the prelude

File: flex_marketplace/src/marketplace/strategy_standard_sale_for_fixed_price.cairo [hunk 2]
```diff
@@ -37,38 +39,36 @@ mod StrategyStandardSaleForFixedPrice {
...
-            contract_address_const::<0>()
+        fn update_protocol_fee(ref self: ContractState, fee: u128) {
+            self.ownable.assert_only_owner();
+            self.protocol_fee.write(fee);
         }
 
         fn protocol_fee(self: @ContractState) -> u128 {
-            // TODO
-            0
+            return self.protocol_fee.read();
```
Comment (ametel01) [inline flex_marketplace/src/marketplace/strategy_standard_sale_for_fixed_price.cairo:71]: no need to explicitly return you can just write `self.protocol_fee.read()`

File: flex_marketplace/src/marketplace/utils/order_types.cairo [hunk 3]
```diff
@@ -1,7 +1,27 @@
+use starknet::ContractAddress;
 #[derive(Drop, Serde)]
-struct MakerOrder { // TODO
+struct MakerOrder {
```
Comment (ametel01) [inline flex_marketplace/src/marketplace/utils/order_types.cairo:3]: this are the definitions of MakerOrder and TakerOrder just copy paste

https://github.com/Flex-NFT-Marketplace/Flex-Marketplace-Contract/blob/e1940e36d8daa8153f1a51c70509adac115f99ea/flex_marketplace/src/marketplace/utils/order_types.cairo

File: flex_marketplace/src/marketplace/strategy_standard_sale_for_fixed_price.cairo [hunk 4]
```diff
@@ -77,14 +77,33 @@ mod StrategyStandardSaleForFixedPrice {
             maker_bid: MakerOrder,
             extra_params: Span<felt252>
         ) -> (bool, u256, u128) {
-            // TODO
-            (true, 0, 0)
+            let priceMatch: bool = maker_bid.price == taker_ask.price;
```
Comment (ametel01) [inline flex_marketplace/src/marketplace/strategy_standard_sale_for_fixed_price.cairo:80]: all the variables must be transformed to snake case eg: `let price_match`

**Review by ametel01 — APPROVED** [review]

**Comment by AniDhumal** [comment]
@ametel01 GTG?

**Comment by ametel01** [comment]
LGTM!

**Pull request merged**